From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Laura Abbott <labbott@redhat.com>
Date: Thu, 2 Jan 2020 14:24:48 -0500
Subject: [PATCH] mptspi: Taint kernel if mptspi is loaded

Author: Tomas Henzl <thenzl@redhat.com>
Date:   Thu Jan 10 16:11:43 2019 -0500

    [message] mptspi: Taint kernel if mptspi is loaded

    Message-id: <20190110161145.5117-2-thenzl@redhat.com>
    Patchwork-id: 237188
    O-Subject: [RHEL-8.0 e-stor 1/3] mptspi: Taint kernel if mptspi is loaded
    Bugzilla: 1651803
    RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
    RH-Acked-by: Jarod Wilson <jarod@redhat.com>
    RH-Acked-by: Chris Leech <cleech@redhat.com>

    RHEL-only
    We are only keeping it to ease developer transitions, we're not
    supporting it, so mark it as unsupported by tainting the kernel and
    issuing an impossible to misunderstand message to the log

    Signed-off-by: Herton R. Krzesinski <herton@redhat.com>

Upstream Status: RHEL only
Signed-off-by: Laura Abbott <labbott@redhat.com>
---
 drivers/message/fusion/mptspi.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/message/fusion/mptspi.c b/drivers/message/fusion/mptspi.c
index eabc4de5816c..c028f5944694 100644
--- a/drivers/message/fusion/mptspi.c
+++ b/drivers/message/fusion/mptspi.c
@@ -1534,6 +1534,12 @@ mptspi_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 		    0, 0, 0, 0, 5);

 	scsi_scan_host(sh);
+
+#ifdef CONFIG_RHEL_DIFFERENCES
+	add_taint(TAINT_SUPPORT_REMOVED, LOCKDEP_STILL_OK);
+	pr_warn("MPTSPI MODULE IS NOT SUPPORTED\n");
+#endif
+
 	return 0;

 out_mptspi_probe:
-- 
2.26.2

